File: vol2/ch23/LeetCode55-1.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        return dfs(nums,nums.size()-1);
    }
    bool dfs(vector<int> &nums,int i) {    //回溯算法
        if(i==0) return true;
        else {
            for(int j=i-1;j>=0; j--) {		//试探位置i-1～0
                if(j+nums[j]>=i)       //出位置j可以跳到位置i
                    return dfs(nums,j);
            }
        }
        return false;
    }
};

File: vol2/ch23/LeetCode55-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        return self.dfs(nums,len(nums)-1)
    def dfs(self,nums,i):       #回溯算法
        if i==0:return True
        else:
            for j in range(i-1,-1,-1):
                if j+nums[j]>=i:    #出位置j可以跳到位置i
                    return self.dfs(nums,j)
        return False
